samples/crypto/sha256/src/main.c

```diff
@@ -73,7 +78,52 @@ int hash_sha256(void)
 	}
 
 	LOG_INF("Hashing successful!");
-	PRINT_HEX("Plaintext", m_plain_text, sizeof(m_plain_text));
+	PRINT_HEX("SHA256 hash", m_hash, sizeof(m_hash));
+
+	return APP_SUCCESS;
+}
+
+int hash_multipart_sha256(void)
+{
+	uint32_t olen;
+	psa_status_t status;
+	uint8_t *input_ptr = m_plain_text;
+	psa_hash_operation_t hash_operation = {0};
+	size_t rest_length_to_hash = sizeof(m_plain_text);
+
+	LOG_INF("Hashing using multi-part SHA256...");
+
+	/* Setup a multipart hash operation */
+	status = psa_hash_setup(&hash_operation, PSA_ALG_SHA_256);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("Could not setup the hash operation! Error %d", status);
+		return APP_ERROR;
+	}
+
+	/* Feed the chunks of the input data to the PSA driver */
+	while (rest_length_to_hash > 0) {
```

Comment:
I am proposing to remove this while all together. Just split the message in two parts and show that. I consider it important for the samples to be very easy to understand. This adds unnecessary complexity in my opinion.

Reply:
Does make the code look better as error checking is required each time, but can remove the loop